src/include/txn.i

```diff
@@ -115,9 +115,9 @@ __wt_txn_oldest_id(WT_SESSION_IMPL *session)
 	 * Take a local copy of these IDs in case they are updated while we are
 	 * checking visibility.
 	 */
-	checkpoint_id = txn_global->checkpoint_id;
-	checkpoint_snap_min = txn_global->checkpoint_snap_min;
-	oldest_id = txn_global->oldest_id;
+	WT_ORDERED_READ(oldest_id, txn_global->oldest_id);
+	WT_ORDERED_READ(checkpoint_gen, txn_global->checkpoint_gen);
```

Comment:
I think it would be sufficient to just do a single `WT_ORDERED_READ` here (on `checkpoint_gen`), but we should think that through.